### ISSUE TITLE
no-tabs: Add an ignoreComments option

### ISSUE THREAD

**Issue by domnantas** [eslint/eslint#12999]
Reviving #11562 
**What rule do you want to change?**
no-tabs

**Does this change cause the rule to produce more or fewer warnings?**
Fewer warnings

**How will the change be implemented? (New option, new default behavior, etc.)?**
Add an `ignoreComments` option

**Please provide some example code that this change will affect:**

When commenting out code which contains tabs:
```js
// {
// \tkey: "value"
// }
```
```js
/*
* {
* \tkey: "value"
* }
*/
```

**What does the rule currently do for this code?**
Throws excessive amount of eslint errors. It is really annoying when it's impossible to comment out code and the only solution is turning off this rule completely.

**What will the rule do after it's changed?**
Allow tabs in comments (useful for codebases using tabs for indentation)

**Are you willing to submit a pull request to implement this change?**
Yes, will resubmit #11563 after confirming it works properly

**Comment by mdjermanovic** [comment]
Makes sense to me :+1: 

There's already an open issue #12438 that includes this + some other options, but there was no activity for a while so I'm not sure how to proceed.

**Comment by mdjermanovic** [comment]
> Yes, will resubmit #11563 after confirming it works properly

Just to clarify, since you mentioned #11563. The option implemented in that PR allows only tabs in _line_ comments, and only in those line comments that have no code before them on the same line.  

Looks like it's targeting whole lines of code commented out with `//`, so it doesn't allow tabs in any `//` after some code, and it doesn't allow tabs in `/* */` at all. 

Is that the proposal, or should the option simply allow all tabs anywhere in any comments:

```js
foo(); // \t calls \t foo \t

/* \t
\t *
* \t tabs \t around \t
\t */ 
```

**Comment by domnantas** [comment]
I think I will close this issue in favor of #12438.

Those are good points about tabs in comments after code - I believe tabs should be allowed in all comments. If I understand correctly, `no-tabs` job is to prevent errors by reporting tabs which may be hinding in actual code, where tabs can cause syntax errors. Tabs in comments won't cause such errors.
